bofire/data_models/features/categorical.py

```diff
@@ -360,37 +359,43 @@ class CategoricalOutput(Output):
     order_id: ClassVar[int] = 8
 
     categories: TCategoryVals
-    objective: Annotated[
-        List[Annotated[float, Field(type=float, ge=0, le=1)]], Field(min_items=2)
-    ]
+    objective: CategoricalObjective = Field(
```

Comment:
I do not understand this change? Why the need to model the desirability of the individual categories by a an own class? Why not keeping the list (ideally transforming it to a tuple?)

Reply:
The challenge is that we need to pass the `CategoricalOutput` type to the `constrained_objective2botorch` to be evaluated by BoTorch as a constraint. To me, this seemed like a more reasonable option as it did not require refactoring the `constrained_objective2botorch` to also look for lists (tuples really). In the end, the objective should produce a scalar of some kind and having the objective just as a list does not allow for such functionality. Maybe I am missunderstanding the question...

Reply:
Ok, makes sense. Then let's do it as proposed by you by an own objective class which holds a tuple of desirability values which are between 0 and 1.